src/main/java/com/cloudogu/externalfile/ExternalFileService.java

```diff
@@ -65,6 +72,26 @@ public void create(Repository repository, CreateExternalFileDto dto) throws IOEx
     }
   }
 
+  public void modify(Repository repository, @Nullable String branch, String path, String url, String commitMessage) throws IOException {
```

Comment:
Why so many parameters? Wouldn't it be easier to pass the whole dto or create a new context object?